File: ProyectoAsisTrack/src/main/java/com/example/ProyectoAsisTrack/Repository/AsignaturasRepository.java
package com.example.ProyectoAsisTrack.Repository;

import com.example.ProyectoAsisTrack.Model.Asignaturas;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface AsignaturasRepository extends JpaRepository<Asignaturas, Long> {
}

File: app/src/main/java/com/example/proyectoasistrack/data/Repository/AsignaturasRepository.kt
package com.example.proyectoasistrack.data.repository

import com.example.proyectoasistrack.data.dao.AsignaturaDao
import com.example.proyectoasistrack.data.model.Asignatura

class AsignaturasRepository(private val dao: AsignaturaDao) {

    fun getAllAsignaturas() = dao.getAllAsignaturas()

    suspend fun insert(asignatura: Asignatura) = dao.insert(asignatura)

    suspend fun update(asignatura: Asignatura) = dao.update(asignatura)

    suspend fun delete(asignatura: Asignatura) = dao.delete(asignatura)
}



